operator/pkg/controller/karmada/controller.go

```diff
@@ -58,41 +58,31 @@ func (ctrl *Controller) Reconcile(ctx context.Context, req controllerruntime.Req
 		if !controllerutil.ContainsFinalizer(karmada, ControllerFinalizerName) {
 			controllerutil.AddFinalizer(karmada, ControllerFinalizerName)
 			if err := ctrl.Update(ctx, karmada); err != nil {
-				return controllerruntime.Result{}, err
+				return controllerruntime.Result{Requeue: true}, err
```

Comment:
I remember the controller will requeue the request to be processed if err is not nil https://github.com/karmada-io/karmada/pull/2973#discussion_r1052886320. I feel like add or not-add is both ok. The key is that we might need to unify the code style in one module which may not confuse others.

Reply:
I agree to that